pkg/client/restclient/client.go

```diff
@@ -38,6 +38,18 @@ const (
 	envBackoffDuration = "KUBE_CLIENT_BACKOFF_DURATION"
 )
 
+// RESTClient for Kubernetes API
+type RESTClientInterface interface {
+	GetRateLimiter() flowcontrol.RateLimiter
+	Verb(verb string) *Request
+	Post() *Request
```

Comment:
It crossed my mind but after counting the number of methods the idea run away from my head very quickly. Still it could have some benefits. Another PR.